umh-core/pkg/config/manager.go

```diff
@@ -686,3 +708,64 @@ func (m *FileConfigManagerWithBackoff) AtomicEditDataflowcomponent(ctx context.C
 
 	return m.configManager.AtomicEditDataflowcomponent(ctx, componentUUID, dfc)
 }
+
+// GetConfigAsString returns the current config file contents as a string
+func (m *FileConfigManager) GetConfigAsString(ctx context.Context) (string, error) {
+	// in the GetConfig method, we already read the file and cached the raw config to m.cacheRawConfig
+	_, err := m.GetConfig(ctx, 0)
+	if err != nil {
+		return "", fmt.Errorf("failed to get config: %w", err)
+	}
+
+	return m.cacheRawConfig, nil
+}
+
+// GetConfigAsString returns the current config as a string with backoff logic for failures
+func (m *FileConfigManagerWithBackoff) GetConfigAsString(ctx context.Context) (string, error) {
+	// in the GetConfig method, we already read the file and cached the raw config to m.cacheRawConfig
+	_, err := m.GetConfig(ctx, 0)
+	if err != nil {
+		return "", fmt.Errorf("failed to get config: %w", err)
+	}
+
+	return m.configManager.cacheRawConfig, nil
+}
+
+// GetConfigModTime returns the modification time of the config file
+func (m *FileConfigManager) GetCacheModTime() (time.Time, error) {
+	// read config to update the cache mod time
+	_, err := m.GetConfig(context.Background(), 0)
```

Comment:
can this function maybe take a context instead? then we can just pass it down here